includes/learndash/class-accredible-learndash-learndash-utils.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Accredible LearnDash Add-on utils class
+ *
+ * @package Accredible_Learndash_Add_On
+ */
+
+defined( 'ABSPATH' ) || die;
+
+require_once plugin_dir_path( __FILE__ ) . '/class-accredible-learndash-learndash-functions.php';
+
+if ( ! class_exists( 'Accredible_Learndash_Learndash_Utils' ) ) :
+	/**
+	 * Accredible LearnDash Add-on utils class.
+	 */
+	final class Accredible_Learndash_Learndash_Utils {
+		/**
+		 * Accredible_Learndash_Learndash_Utils constructor.
+		 *
+		 * @param Accredible_Learndash_Learndash_Functions $functions A mock instance to be passed when unit testing.
+		 */
+		public function __construct( $functions = null ) {
+			if ( null !== $functions ) {
+				$this->functions = $functions;
+			} else {
+				$this->functions = new Accredible_Learndash_Learndash_Functions();
+			}
+		}
```

Comment:
I found mocking objects in PHPUnit quite challenging unlike Rspec in Ruby. I tried to use [Mockery](https://github.com/mockery/mockery) but it didn't work well with our codebase. So this design is the simplest way that I could think of.